mergify_engine/web/api/queues.py

```diff
@@ -317,5 +324,406 @@ async def repository_queues(
                 )
 
             queues.queues.append(queue)
-
         return queues
+
+
+QueuePauseSchema = voluptuous.Schema({voluptuous.Required("reason"): str})
+
+
+@router.post(
+    "/repos/{owner}/{repository}/queue/{queue_name}/pause-from",  # noqa: FS003
+    summary="Pauses merge queue(s)",
+    description="Pauses the merge of a queue and the queues following it",
+    responses={
+        **api.default_responses,  # type: ignore
+        404: {"description": "Not found"},
+        200: {"description": "The queue(s) have been paused"},
+    },
+)
+async def pause_queues_from(
+    request: requests.Request,
+    application: application_mod.Application = fastapi.Depends(  # noqa: B008
+        security.get_application
+    ),
+    owner: github_types.GitHubLogin = fastapi.Path(  # noqa: B008
+        ..., description="The owner of the repository"
+    ),
+    repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the repository"
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
+        redis.get_redis_cache
+    ),
+    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
+        redis.get_redis_queue
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+) -> responses.JSONResponse:
+
+    try:
+        raw_json = await request.json()
+        data = QueuePauseSchema(raw_json)
+
+    except voluptuous.Invalid:
+        return responses.JSONResponse(
+            status_code=400, content={"error": "Invalid json"}
+        )
+
+    reason = data["reason"]
+    if reason == "":
+        reason = "No pause reason was specified"
+
+    async with github.aget_client(installation_json) as client:
+        try:
+            # Check this token as access to this repository
+            repo = typing.cast(
+                github_types.GitHubRepository,
+                await client.item(f"/repos/{owner}/{repository}"),
+            )
+        except (http.HTTPNotFound, http.HTTPForbidden, http.HTTPUnauthorized):
+            raise fastapi.HTTPException(status_code=404)
+
+        sub = await subscription.Subscription.get_subscription(
+            redis_cache, repo["owner"]["id"]
+        )
+
+        installation = context.Installation(
+            installation_json, sub, client, redis_cache, redis_queue
+        )
+        repository_ctxt = installation.get_repository_from_github_data(repo)
+        config_file = await repository_ctxt.get_mergify_config_file()
+
+        if config_file and installation.redis_queue:
```

Comment:
check for `redis_queue` before? there's no point in getting the `config_file` if you don't use it if `redis_queue` is None.

And I'd tend to use `is not None` to be clear of what you're checking